--- libtool-2.4.6/libltdl/ltdl.c~	2015-01-16 18:52:04.000000000 +0000
+++ libtool-2.4.6/libltdl/ltdl.c	2018-09-05 08:55:37.814901898 +0000
@@ -79,6 +79,8 @@
 static  const char	libprefix[]		= LT_LIBPREFIX;
 #if defined LT_MODULE_EXT
 static	const char	shlib_ext[]		= LT_MODULE_EXT;
+static	const char	shlib_ext_avx2[]		= ".so.avx2";
+static	const char	shlib_ext_avx512[]		= ".so.avx512";
 #endif
 /* If the loadable module suffix is not the same as the linkable
  * shared library suffix, this will be defined. */
@@ -1688,6 +1690,29 @@
 	return handle;
 
 #if defined LT_MODULE_EXT
+      if (__builtin_cpu_supports("avx512dq")) {
+	      /* Try appending SHLIB_EXT.   */
+	      LT__SETERRORSTR (saved_error);
+	      errors = try_dlopen (&handle, filename, shlib_ext_avx512, advise);
+
+	      /* As before, if the file was found but loading failed, return now
+		 with the current error message.  */
+	      if (handle || ((errors > 0) && !file_not_found ()))
+		return handle;
+      }
+
+      if (__builtin_cpu_supports("avx2")) {
+	      /* Try appending SHLIB_EXT.   */
+	      LT__SETERRORSTR (saved_error);
+	      errors = try_dlopen (&handle, filename, shlib_ext_avx2, advise);
+
+	      /* As before, if the file was found but loading failed, return now
+		 with the current error message.  */
+	      if (handle || ((errors > 0) && !file_not_found ()))
+		return handle;
+      }
+
+
       /* Try appending SHLIB_EXT.   */
       LT__SETERRORSTR (saved_error);
       errors = try_dlopen (&handle, filename, shlib_ext, advise);
